File: rust/src/solutions/euler012.rs
use crate::utils::numeric::divisors::divisors;
use crate::utils::numeric::sequences::triangular::Triangular;

pub fn euler012() -> String {
    Triangular::<i64>::new()
        .find(|&x| divisors(x).len() > 500)
        .unwrap()
        .to_string()
}

File: clojure/src/project_euler/solutions/euler012.clj
(ns project-euler.solutions.euler012
  (:require [project-euler.utils.numeric :refer [triangular]])
  (:require [project-euler.utils.primes :refer [divisors]]))

(defn euler012
  []
  (str (first (filter #(> (count (divisors %)) 500) (triangular)))))